client/src/components/reservations/ReservationForm.jsx

```diff
@@ -0,0 +1,201 @@
+import React, { useState } from "react";
+import { useGetRoomById } from "../../hooks/useGetRoomById";
+import { Navigate, useParams } from "react-router-dom";
+import { useCreateReservation } from "../../hooks/useCreateReservation";
+import { ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+export const ReservationForm = () => {
+  const { id } = useParams();
+
+  // eslint-disable-next-line
+  const {loading,reservedDates,room}=useGetRoomById(id)
+
+  const [completed, setCompleted] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const { reservation, createReservation } = useCreateReservation();
+
+  const [arrivalDate, setArrivalDate] = useState("");
+  const [departureDate, setDepartureDate] = useState("");
+  const [firstLastName, setFirstLastName] = useState("");
+  const [numberOfPersons, setNumberOfPersons] = useState("");
+  const [email, setEmail] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [additionalInfo, setAdditionalInfo] = useState("");
+  const [roomId, setRoomId] = useState(id);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const phoneRegex = /^\+(?:[0-9] ?){6,14}[0-9]$/;
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+    if (!arrivalDate || !departureDate) {
+      setErrorMessage("Molimo Vas da izaberete datume dolaska i odlaska.");
+      return;
+    }
+
+    if (departureDate <= arrivalDate) {
+      setErrorMessage("Datum odlaska mora biti posle datuma dolaska.");
+      return;
+    }
+
+    if (!phoneRegex.test(phoneNumber)) {
+      setErrorMessage("Molimo unesite ispravan format broja telefona.");
+      return;
+    }
+
+    if (firstLastName.length < 6) {
+      setErrorMessage("Molimo unesite puno ime i prezime.");
+      return;
+    }
+
+    if (!emailRegex.test(email)) {
+      setErrorMessage("Molimo unesite ispravan format e-pošte.");
+      return;
+    }
+
+    if (!numberOfPersons || !email || !phoneNumber || !firstLastName) {
+      setErrorMessage("Molimo popunite sva polja.");
+      return;
+    }
+
+    const isDateAvailable =
```

Comment:
What if you need to check for date availability in other processes?
What if you need to check date availability in an online booking process via email, not the web page?

Reservation checking should be moved to backend. Suggested refactoring: create a server endpoint to check for date availability, then you just call an api endpoint here in the component (via reservationApi) and voila, you have a consistent process of checking date availability not baked into a single react component. Remember:  it's important to separate business logic (check date) from your rendering/frontend as much as possible

Reply:
Done